src/components/Property/RaiseRepair/RateScheduleItemView.js

```diff
@@ -31,16 +34,55 @@ const RateScheduleItemView = ({
     return sorCodes.filter((a) => a.code == value)[0]
   }
 
+  const updateTotalCost = (rateScheduleItemCosts) => {
+    setTotalCost(calculateTotalCost(rateScheduleItemCosts, 'cost', 'quantity'))
+  }
+
+  const onQuantityInput = (index, event) => {
+    const quantity = parseFloat(event.target.value) || 0
+    const costPerUnit = parseFloat(
+      document.getElementById(`rateScheduleItems[${index}][cost]`).value
+    )
+
+    const rateScheduleItemCostAtSameIndex = rateScheduleItemCosts.find(
+      (e) => e.index === index
+    )
+
+    if (rateScheduleItemCostAtSameIndex) {
+      rateScheduleItemCostAtSameIndex.cost = costPerUnit
+      rateScheduleItemCostAtSameIndex.quantity = quantity
+
+      setRateScheduleItemCosts(rateScheduleItemCosts)
+      updateTotalCost(rateScheduleItemCosts)
+    } else {
+      const newRateScheduleItemCosts = [
+        ...rateScheduleItemCosts,
+        {
+          index: index,
```

Comment:
Do we need an index here if we're using an array? Wondering if we can just index into it like 
`rateScheduleItemCosts[index] =  { cost: costPerUnit, quantity: quantity }`

Reply:
If that's applicable I'm not sure exactly how that fits with our state hook method though..

Reply:
Ah this would be a lot simpler... Let me give it a go

Reply:
I think it becomes tricky when managing the state of adding and removing the ​RateScheduleItem components.
If i'm not keeping track of the index, then when using `slice()` to remove one of the elements from the array, the indexes change to be incremental again. But the way the multi components work is that there may be a RateScheduleItem with index 0 and index 4 with (1..3) removed. I'd need a way of targeting the index at 4 but this new array would just look like, for example:
```
0: {cost: 74.89, quantity: 3}
1: {cost: 24.89, quantity: 2}
```

Do you get what I mean - sorry if i'm not communicating this well. Hard to explain in words here - let me know if you want a better explanation. Happy if you wanted to fiddle around with the code yourself to make your simpler suggestion work when taking into consideration adding/removing SOR codes.

Reply:
I'm going to merge this for now (with all your other comments addressed) but I'd like to revisit this and make it work with your nicer suggestion above